src/nvim/rbuffer.c

```diff
@@ -34,17 +34,20 @@ void rbuffer_free(RBuffer *buf)
   xfree(buf);
 }
 
-size_t rbuffer_size(RBuffer *buf) FUNC_ATTR_NONNULL_ALL
+size_t rbuffer_size(RBuffer *buf)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
 {
   return buf->size;
 }
 
-size_t rbuffer_capacity(RBuffer *buf) FUNC_ATTR_NONNULL_ALL
+size_t rbuffer_capacity(RBuffer *buf)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
 {
   return (size_t)(buf->end_ptr - buf->start_ptr);
 }
 
-size_t rbuffer_space(RBuffer *buf) FUNC_ATTR_NONNULL_ALL
+size_t rbuffer_space(RBuffer *buf)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
```

Comment:
Maybe the compiler will consider the memory as changed if we use locks to access `RBuffer`(which we do, if I remember correctly), but then this attribute won't be of much help. In any case, since the computation is very trivial, it is better to be safe than sorry(trying to debug it in a far future, where we forgot about this)